runhouse/main.py

```diff
@@ -679,6 +303,196 @@ def cluster_list(
+            )
+    except Exception as e:
+        console.print(f"Failed to keep the cluster warm: {e}")
+
+
+@cluster_app.command("up")
+def cluster_up(
+    cluster_name: str = typer.Argument(
+        None,
+        help="The name of cluster to bring up.",
+    )
+):
+    """Bring up the cluster if it is not up. No-op if cluster is already up.
+    This only applies to on-demand clusters, and has no effect on self-managed clusters.
+
+    Example:
+
+        ``$ runhouse cluster up /sashab/rh-basic-cpu``
+
+    """
+    current_cluster = get_cluster_or_local(cluster_name)
+
+    try:
+        current_cluster.up_if_not()
+    except ValueError:
+        console.print("Cluster is not saved in den.")
+    except Exception as e:
+        console.print(f"Failed to bring up the cluster: {e}")
+
+
+@cluster_app.command("down")
+def cluster_down(
+    cluster_name: str = typer.Argument(
+        None,
+        help="The name of cluster to terminate.",
+    ),
+    remove_configs: bool = typer.Option(
+        False,
+        "-rm",
+        "--remove-configs",
+        help="Whether to delete cluster config from Den (default: ``False``).",
+    ),
+    remove_all: bool = typer.Option(
+        False,
+        "-a",
+        "--all",
+        help="Whether to terminate all running clusters saved in Den (default: ``False``).",
+    ),
+    confirm_deletion: bool = typer.Option(
+        False, "-y", "--yes", help="Skip confirmation prompt."
+    ),
+):
+    """Terminate cluster if it is not down. No-op if cluster is already down.
+    This only applies to on-demand clusters, and has no effect on self-managed clusters.
+
+    Example:
+
+        ``$ runhouse cluster down /sashab/rh-basic-cpu``
+    """
+    if not confirm_deletion:
+        if cluster_name:
+            proceed = typer.prompt(f"Terminating: {cluster_name}. Proceed? [Y/n]")
+        elif remove_all:
+            proceed = typer.prompt(
+                "Terminating all running clusters saved in Den. Proceed? [Y/n]"
+            )
+        else:
+            console.print("Cannot determine which cluster to terminate, aborting.")
+            raise typer.Exit(1)
+
+        if proceed.lower() == "n":
+            console.print("Aborted!")
+            raise typer.Exit(0)
+
+    if remove_all:
+        running_den_clusters = Cluster.list(status=ClustersListStatus.running).get(
+            "running_clusters"
+        )
+
+        if not running_den_clusters:
+            console.print("No running clusters saved in Den")
+            raise typer.Exit(0)
+
+        terminated_clusters: int = 0
+        for running_cluster in running_den_clusters:
+            try:
+                current_cluster = rh.cluster(
+                    name=f'/{rns_client.username}/{running_cluster.get("Name")}'
+                )
+                current_cluster.teardown_and_delete() if remove_configs else current_cluster.teardown()
```

Comment:
we can only call `teardown` and `teardown_and_delete` if they are on-demand clusters, we should check that first otherwise we'll get errors:

```
AttributeError: 'Cluster' object has no attribute 'teardown'
```